atlasaction/action.go

```diff
@@ -118,22 +126,41 @@ func MigrateLint(ctx context.Context, client *atlasexec.Client, act *githubactio
 		Context:   string(buf),
 		Web:       true,
 		Writer:    &resp,
+		Format:    `{"URL": "{{ .URL }}", "DiagnosticsCount": {{ .DiagnosticsCount }}, "FilesCount": {{ len .Files }}}`,
 	})
-	url := strings.TrimSpace(resp.String())
-	if url != "" {
-		act.SetOutput("report-url", url)
-		if publishErr := publishResult(url, err, act); publishErr != nil {
-			act.Warningf("unable to publish lint report: %v", publishErr)
-		}
+	if err != nil && !errors.Is(err, atlasexec.LintErr) {
+		return err
 	}
-	return err
+	if err := json.NewDecoder(&resp).Decode(&payload); err != nil {
+		return fmt.Errorf("decoding payload: %w", err)
+	}
+	if payload.URL != "" {
+		act.SetOutput("report-url", payload.URL)
+	}
+	dirName := act.GetInput("dir-name")
+	var summary bytes.Buffer
+	if err := comment.Execute(&summary, payload); err != nil {
+		return err
+	}
+	if err := publish(act, dirName, summary.String()); err != nil {
+		return err
+	}
+	if errors.Is(err, atlasexec.LintErr) {
+		return fmt.Errorf("lint completed with errors, see report: %s", payload.URL)
+	}
+	return nil
 }
 
-func publishResult(url string, err error, act *githubactions.Action) error {
-	status := "success"
-	if err != nil {
-		status = "error"
-	}
+var (
+	//go:embed comment.tmpl
+	commentTmpl string
+	comment     = template.Must(template.New("comment").Parse(commentTmpl))
```

Comment:
Just a thought, not sure it's easy: can you put the commentMarker inside the template?